x-pack/plugins/security_solution/public/timelines/components/timeline/tabs/esql/index.tsx

```diff
@@ -126,8 +137,9 @@ export const DiscoverTabContent: FC<DiscoverTabContentProps> = ({ timelineId })
     if (!index) return;
     if (!latestState || combinedDiscoverSavedSearchStateRef.current === latestState) return;
     if (isEqualWith(latestState, savedSearchById, savedSearchComparator)) return;
-    updateSavedSearch(latestState, timelineId);
-    combinedDiscoverSavedSearchStateRef.current = latestState;
+    updateSavedSearch(latestState, timelineId, () => {
```

Comment:
```suggestion
    updateSavedSearch(latestState, timelineId, function onUpdate() {
```

Since the parameters of `updateSavedSearch` are not named, consider naming this function to easier make sense of this later on.